app/services/concerns/xccdf_report/rule_references.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module XCCDFReport
+  # Methods related to saving rule references and finding which rules
+  # they belong to
+  module RuleReferences
+    extend ActiveSupport::Concern
+
+    included do
+      def rule_references
+        @rule_references ||= new_rules.map do |rule|
+          RuleReference.from_oscap_objects(rule.references)
+        end
+      end
+
+      def save_rule_references
+        RuleReference.import(new_rule_references,
+                             columns: %i[href label],
+                             ignore: true)
+      end
+
+      def associate_rule_references(new_rules)
+        new_rules.zip(@rule_references || []).each do |rule, references|
```

Comment:
Setting `@rule_references` here (in the old code with `||=`) does not seem right to me. Using this `@rule_references || []` accomplishes the same thing without polluting the instance variable.